bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,722 @@
+        }
+    }
+
+    private void updateAllChannels() {
+        if (disposed) {
+            return;
+        }
+        if (refreshCounter > 0) {
+            refreshCounter--;
+            if (refreshCounter == 0) {
+                fetchSchedule();
+                return;
+            }
+            if (refreshCounter % 5 == 1) {
+                parseImage();
+                updateState(new ChannelUID(getThing().getUID(), CHANNEL_CAMERA), currentImage);
+                return;
+            }
+        }
+        String status = null;
+        String url = buildHttpAddress("/status.txt");
+        try {
+            status = HttpUtil.executeUrl("GET", url, 1000);
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+        if (status != null && !status.isEmpty()) {
+            boolean parsingOk = true;
+            String[] rows = status.split("\\r?\\n");
+            ChannelUID channel;
+            for (String row : rows) {
+                int idx = row.indexOf('=');
+                if (idx == -1) {
+                    continue;
+                }
+                final String key = row.substring(0, idx);
+                String value = row.substring(idx + 1).replace("\"", "");
+                switch (key) {
+                    case "JSON_ROBOT_STATE":
+                        if (value.isEmpty()) {
+                            value = HBSTATE_UNKNOWN;
+                        }
+                        if (!value.equals(currentState)) {
+                            currentState = value;
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_STATE);
+                            updateState(channel, StringType.valueOf(value));
+
+                            if (value.equals(HBSTATE_WORKING) || value.equals(HBSTATE_BACKMOVING)
+                                    || value.equals(HBSTATE_BACKMOVING_INIT)) {
+                                currentCleanState = OnOffType.ON;
+                                currentStartState = OnOffType.ON;
+                                currentHomeState = OnOffType.OFF;
+                            } else if (value.equals(HBSTATE_HOMING) || value.equals(HBSTATE_DOCKING)) {
+                                currentCleanState = OnOffType.OFF;
+                                currentStartState = OnOffType.OFF;
+                                currentHomeState = OnOffType.ON;
+                            } else {
+                                currentCleanState = OnOffType.OFF;
+                                currentStartState = OnOffType.OFF;
+                                currentHomeState = OnOffType.OFF;
+                            }
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_CLEAN);
```

Comment:
You can just pass the `channelID` no reason to create the channel object first

Reply:
Please apply where applicable saves a lot of code :-)

Reply:
Where do I get channelID from then?

Reply:
> Where do I get channelID from then?
Sorry, didn't see the String version...

I see that you found it. Indeed with `channelID` i referred to the String which you pass as the second argument to the ChannelUID constructor.

Reply:
Yeah that was much better :smile: